src/ResizeObserverController.js

```diff
@@ -151,28 +174,40 @@ export default class ResizeObserverController {
             return;
         }
 
+        const rootNode = this.rootNode_;
+        const doc = rootNode.ownerDocument || rootNode;
+        const win = doc.defaultView;
+
         // Subscription to the "Transitionend" event is used as a workaround for
         // delayed transitions. This way it's possible to capture at least the
         // final state of an element.
-        document.addEventListener('transitionend', this.onTransitionEnd_);
+        rootNode.addEventListener('transitionend', this.onTransitionEnd_, true);
 
-        window.addEventListener('resize', this.refresh);
+        if (win) {
+            win.addEventListener('resize', this.refresh, true);
+        }
 
         if (mutationObserverSupported) {
             this.mutationsObserver_ = new MutationObserver(this.refresh);
 
-            this.mutationsObserver_.observe(document, {
+            this.mutationsObserver_.observe(rootNode, {
                 attributes: true,
                 childList: true,
                 characterData: true,
                 subtree: true
             });
         } else {
-            document.addEventListener('DOMSubtreeModified', this.refresh);
+            rootNode.addEventListener('DOMSubtreeModified', this.refresh, true);
```

Comment:
FMI: Is this adding the `capture` option so that no subnodes receive the event as a performance optimization? How  do we know that none of the subnodes care about the event?

Reply:
`capture` is to ensure that no one in the DOM hierarchy stops propagation before we receive the event. So, we don't really need to know whether subnodes handle the event or not.